src/AI/CapPhaseAnalyzer.java

```diff
@@ -0,0 +1,454 @@
+        inf.x = ownedFac.xCoord;
+        inf.y = ownedFac.yCoord;
+
+        final GamePath infPath = findFeasiblePath(inf, propXYC, map);
+        if( null == infPath || infPath.getPathLength() < 1 )
+          continue; // Can't reach this city
+
+        int oldDistance = Integer.MAX_VALUE;
+        if( possibleOwners.containsKey(owner.army) )
+          oldDistance = possibleOwners.get(owner.army);
+
+        int distance = infPath.getFuelCost(inf, map) * 100 / inf.getMovePower(map);
+        if( distance < oldDistance )
+          possibleOwners.put(owner.army, distance);
+      }
+
+
+      // Calculate who's the closest, and if that army has real competition for this prop
+      Army closestArmy = null;
+      int closestDistance = Integer.MAX_VALUE;
+      for( Army army : possibleOwners.keySet() )
+      {
+        int distance = possibleOwners.get(army);
+        if( distance < closestDistance )
+        {
+          closestArmy = army;
+          closestDistance = distance;
+        }
+      }
+
+      boolean contested = false;
+      for( Army army : possibleOwners.keySet() )
+      {
+        if(!army.isEnemy(closestArmy))
+          continue;
+        int distance = possibleOwners.get(army);
+        int distanceDelta = Math.abs(closestDistance - distance);
+        contested |= ( distanceDelta <= 100 );
+      }
+
+      // If it isn't contested and we want to try to cap it, add it to the list
+      if( contested )
+        contestedProps.add(propXYC);
+      else if( viewer == closestArmy
+            && viewer.isEnemy(map.getLocation(propXYC).getOwner()) ) // Don't try to cap it if we already own it
+        rightfulProps.add(propXYC);
+    }
+
+    // Build cap chains to factories; don't continue them, since cap chains from that factory will be considered separately
+    ArrayList<ArrayList<CapStop>> factoryCapChains = new ArrayList<>();
+    while (!rightfulFactories.isEmpty())
+    {
+      final XYCoord dest = rightfulFactories.remove(0);
+      Utils.sortLocationsByDistance(dest, startingFactories);
+      final XYCoord start = startingFactories.get(0);
+      final Commander owner = factoryOwnership.get(start);
+
+      final Unit inf = new Unit(owner, owner.getUnitModel(UnitModel.TROOP));
+      inf.x = start.xCoord;
+      inf.y = start.yCoord;
+
+      final GamePath infPath = findFeasiblePath(inf, dest, map);
+      if( null == infPath || infPath.getPathLength() < 1 )
+        continue; // Can't reach
+
+      int distance = infPath.getFuelCost(inf, map);
+
+      ArrayList<CapStop> chain = new ArrayList<>();
+      CapStop build = new CapStop(start);
+      build.extraTurns = distance/inf.getMovePower(map) - 13;
+      chain.add(build);
+      CapStop cap = new CapStop(dest);
+      chain.add(cap);
+      factoryCapChains.add(chain);
+
+      // Now that we're just in cap-chain land, don't worry about whether we start with this factory or not.
+      startingFactories.add(dest);
```

Comment:
Huh. Interesting detail. So cap chains are planned from factories that are "rightfully" ours even if we don't have them yet, and just... hang around until infantry get built from those factories to take the assignment.

Reply:
Technically, the infantry that snagged the factory will be doing the first chain.
But yes, this cap phase stuff definitely assumes the map is set up without contested production (and I'm not really sure how I *would* plan around contested production).

There's probably room here to modify the cap chain growth loop to account for these factories' chains starting later, but the added complexity probably isn't worth it.
Especially since this code should probably be expanded to handle bikes at some point...

Reply:
Couldn't this end up prioritizing paths from uncaptured factories to uncaptured bulidings that are more efficiently captured by an already-owned factory?

Reply:
Technically yes.
That's what I meant by `There's probably room here to modify the cap chain growth loop to account for these factories' chains starting later, but the added complexity probably isn't worth it.`

Reply:
Speaking of contested... if a cap chain was built from a "rightful" factory, and then it was contested, and then halfway through the game we finally claimed it... would the cap chain just hang around until that factory built its first infantry? Or I guess the infantry that captured it could be immediately assigned that cap chain. Is there any statute of limitations on cap chains?

Reply:
Yeah, that could happen.
JakeMan (and by extension this cap logic, which perhaps is worth fixing) is designed around conventional PvP scenarios.

If you load a save during a game, you'll also end up re-generating cap chains to any unclaimed/uncontested properties, which is also questionable.
I want to introduce some controlled state-saving for AIs down the line to fix that and similar stuff.
I guess we could just call the whole AI class `Serializable` and call it a day, but that feels questionable for some reason I can't quite put my finger on.